### PR TITLE
copr: support case when and coalesce

### PR DESCRIPTION
**Pull request by BusyJay**
ref #1226 

@coocood @siddontang PTAL

### CODE REVIEW

File: src/util/xeval/evaluator.rs [hunk 1]
```diff
@@ -303,6 +305,31 @@ impl Evaluator {
         let (left, right) = try!(self.eval_two_children(ctx, expr));
         eval_arith(left, right, f)
     }
+
+    fn eval_case_when(&mut self, ctx: &EvalContext, expr: &Expr) -> Result<Datum> {
+        for chunk in expr.get_children().chunks(2) {
+            let res = try!(self.eval(ctx, &chunk[0]));
+            if chunk.len() == 1 {
```
Comment (siddontang) [inline src/util/xeval/evaluator.rs]: seem to be different with https://github.com/pingcap/tidb/pull/1856/files#diff-9c6a66f0086282f3806a4c7ed4eecf05R44?

Comment (BusyJay) [inline src/util/xeval/evaluator.rs]: Actually they are same. TiDB loops over the collection by step 2; here use chunks. And the loop in TiDB will end if the size is an odd number; chunks will still pass the last element in that case.

**Comment by siddontang** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**